File: Project3/task2.cpp
/*
* searches for a value in sorted array
*   arr is an array to search in
*   value is searched value
*   left is an index of left boundary
*   right is an index of right boundary
* returns position of searched value, if it presents in the array
* or -1, if it is absent
*/

#include <iostream>
int binarySearch(int arr[], int value) {
      int left=1;
      int right=sizeof(arr);
      while (left <= right) {
            int middle = (left + right) / 2;
            if (arr[middle] == value)
                  return middle;
            else if (arr[middle] > value)
                  right = middle - 1;
            else
                  left = middle + 1;
      }
      return -1;

}

int main (void) {

  int arr[]={1,2,3,4,5,6,7};
  std::cout << "List for executing binary search: ";
  for(int i=0;i<(sizeof(arr)/sizeof(*arr));i++){
      std::cout << arr[i];
  }
  std::cout << "\n"<<"Executing binary search on 5... " << binarySearch(arr,5) <<"\n";
  std::cout << "Exectuing binary search on 10..." << binarySearch(arr,10)<<"\n";
  return 0;
}

File: Project3/task2.rb
def binary_search (a,key)  # a is the array and key is the value to be found if key not found, returns "value not found in the array"
    lo = 0
    hi= a.length-1
    
    while (lo<=hi)
        mid = lo+((hi-lo)/2)
        
        if a[mid] == key
            return mid
        elsif a[mid] < key
            lo=mid+1
        else
            hi=mid-1
        end
        
    end
    
    return "Value not found in the array"
end

list = [1,2,3,4,5,6,7]
puts "List for binary searching: #{list}"
puts "Executing binary search for 6...#{binary_search(list,6)}"
puts "Executing Binary search for 8...#{binary_search(list,10)}"